### PR TITLE
Validate change_dataset's --endpoint-url argument

### PR DESCRIPTION
**Pull request by lbam**
The argument needs to contain `{table_id}` for DSO-API to insert the actual table name.

### CODE REVIEW

**Review by jjmurre — APPROVED** [review]
LGTM. Good idea to add this extra precaution.

File: src/schematools/contrib/django/management/commands/change_dataset.py [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
+from typing import Any
 
 from django.core.exceptions import ValidationError
 from django.core.management import BaseCommand, CommandError
 
 from schematools.contrib.django.models import Dataset
 
 
-def _strtobool(value):
+def _strtobool(value: Any) -> bool:
```
Comment (gkoller) [inline src/schematools/contrib/django/management/commands/change_dataset.py:12]: Judging by its name `strtobool` this function converts a `str` to a `bool`. The docstring of the function you delegate to confirm that: _Convert a string representation of truth to true (1) or false (0)._. Hence the type annotation of `value` should not be `Any`, but `str`

File: src/schematools/contrib/django/management/commands/change_dataset.py [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
     try:
         return bool(strtobool(value))
     except ValueError:
         raise ArgumentTypeError("expected boolean value") from None
```
Comment (gkoller) [inline src/schematools/contrib/django/management/commands/change_dataset.py:16]: I realize this is existing code, but the message is incorrect. A boolean value was NOT expected. On the contrary the boolean value is what this function is supposed to be generating. An arguably better message would be: _Expected a truth value_. Which is basically the message of the `ValueError` exception. Hence my suggestion would be to change this into:

```python
except ValueError as ve:
            raise ArgumentTypeError(str(ve)) from None
```

File: src/schematools/contrib/django/management/commands/change_dataset.py [hunk 3]
```diff
@@ -69,13 +71,19 @@ def handle(self, *args, **options):
             raise CommandError(f"Dataset not found: {name}.\nAvailable are: {available}") from None
 
         # Validate illogical combinations
-        if options.get("endpoint_url"):
+        endpoint_url = options.get("endpoint_url")
+        if endpoint_url is not None:
```
Comment (gkoller) [inline src/schematools/contrib/django/management/commands/change_dataset.py:75]: More idiomatic:

```python
if endpoint_url:
    ..
```.

File: src/schematools/contrib/django/management/commands/change_dataset.py [hunk 4]
```diff
@@ -69,13 +71,19 @@ def handle(self, *args, **options):
...
             if options.get("enable_db"):
                 raise CommandError("Can't use --endpoint-url with --enable-db")
 
             if dataset.enable_db:
                 options["enable_db"] = False
 
+        # URL endpoints need to contain {table_id}, which DSO-API replaces
+        # with the actual table name.
+        if endpoint_url is not None and "{table_id}" not in endpoint_url:
```
Comment (gkoller) [inline src/schematools/contrib/django/management/commands/change_dataset.py:84]: More idiomatic:

```python
if endpoint_url and "{table_id}" not in endpoint_url:
    ...
```

File: src/schematools/contrib/django/management/commands/change_dataset.py [hunk 5]
```diff
@@ -69,13 +71,19 @@ def handle(self, *args, **options):
...
 
         # Validate illogical combinations
-        if options.get("endpoint_url"):
+        endpoint_url = options.get("endpoint_url")
+        if endpoint_url is not None:
             if options.get("enable_db"):
                 raise CommandError("Can't use --endpoint-url with --enable-db")
 
             if dataset.enable_db:
                 options["enable_db"] = False
```
Comment (gkoller) [inline src/schematools/contrib/django/management/commands/change_dataset.py:80]: No need for an if-statement:

```python
options["enable_db"] = dataset.enable_db
```

Comment (gkoller) [reply src/schematools/contrib/django/management/commands/change_dataset.py:80]: Existing code, I know, but no harm in cleaning up code when we (almost) hit it.

File: src/schematools/contrib/django/management/commands/change_dataset.py (hunk 1 above)
Comment (lbam) [reply src/schematools/contrib/django/management/commands/change_dataset.py:12]: Fixed in 94e532695932246f7fa2a57e090f18b01043a59d.

**Pull request merged**